lib/exhal/navigation.ex

```diff
@@ -25,13 +25,13 @@ defmodule ExHal.Navigation do
   @doc """
   Follows all links of a particular rel in a HAL document.
 
-  Returns `[{:ok, %ExHal.Document{...}, %ExHal.ResponseHeader{...}}, {:error, %ExHal.Error{...}, ...]`
+  Returns `[{:ok, %ExHal.Document{...}, %ExHal.ResponseHeader{...}}, {:error, %ExHal.NoSuchLinkError{...}, {:error, %ExHal.Error{...} ...]`
```

Comment:
I wonder if `NoSuchLinkError` should be embedded in a `List`. It is not the same as failing to fetch one target of a compound link. If the document doesn't have a link that's it.

I see a couple of reasonable choices when the link doesn't exist:
 1. return an empty list
 2. return an error tuple like `{:error, %ExHal.NoSuchLinkError{...}}`

If we are going to make a breaking change to the API (which this PR does) we should do one of those things.

Reply:
I opted to go with `{:error, %ExHal.NoSuchLinkError{...}}` and added test coverage